### PR TITLE
Remove suppport for Hadoop 3.2

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Related to https://github.com/stackabletech/docker-images/pull/571

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by sbernauer** [comment]
Full (not nightly) testsuite: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hdfs-operator-it-custom/125/

File: rust/operator-binary/src/hdfs_controller.rs [hunk 1]
```diff
@@ -195,11 +195,6 @@ pub enum Error {
         source: stackable_operator::error::Error,
     },
 
-    #[snafu(display(
-        "kerberos not supported for HDFS versions < 3.3.x. Please use at least version 3.3.x"
-    ))]
-    KerberosNotSupported,
-
```
Comment (NickLarsenNZ) [inline rust/operator-binary/src/hdfs_controller.rs:202]: :raised_hands:

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Comment by sbernauer** [comment]
Next try https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hdfs-operator-it-custom/127/

**Comment by sbernauer** [comment]
Tests passed

**Pull request merged**